go2ts.go

```diff
@@ -96,38 +128,90 @@ func (g *Go2TS) tSTypeFromStructFieldType(reflectType reflect.Type) *tsType {
 		kind = reflectType.Kind()
 	}
 
-	// Default to setting the tsType from the Go type.
-	ret.typeName = reflectTypeToTypeScriptType(reflectType)
+	// As we build up the chain of tsTypes that fully describes a typeDefinition
+	// we may come across named types. For example: map[string]Donut, where
+	// Donut could be a "type Donut struct {...}", or a type based on a
+	// primitive type, such as "type Donut string". In this case we need to add
+	// that type to all of our known types and return a reference to that type
+	// from here.
+	if !calledFromAddType && // Don't do this if called from addType().
+		reflectType.Name() != "" && // Don't bother with anonymous structs.
+		!isTime(reflectType) && // Also skip time.Time.
+		(!isPrimitive(reflectType.Kind()) || // And either it's not a primitive Kind.
+			// Or it's case where a primitive Kind like string shows up with a type name.
+			(isPrimitive(reflectType.Kind()) && reflectType.Name() != reflectType.Kind().String())) {
```

Comment:
It seems like only string type aliases (e.g. `type Donut string`) get this special treatment. What about type aliases for other primitive types?

Reply:
This conditional is for all primitive types.